assignments/Joyce/week4-hw-joyce.js

```diff
@@ -0,0 +1,85 @@
+// Question 1: write a function `circleArea` which have one argument `radius`
+// and will return circlearea given `radius`
+
+function circleArea(radius) {
+   return Math.PI * radius * radius
+}
+
+// console.log(circleArea(5))
+
+// ---
+// Question 2: write a function `compact` which have one argument `ary` and
+// returns a copy of `ary` with all null elements removed.
+// For example `compact([1, null, false, 3, null, 2])`` should return [1, false, 3, 2]
+
+function compact(ary) {
+	let ary1 = []
+	for (let i = 0; i < ary.length; i++){
+		if (ary[i] != null){
+			ary1.push(ary[i])
+		}
+	}
+	return ary1
+}
+
+// console.log(compact([1, null, false, 3, null, 2]))
+
+// ---
+// Question 3: write a function `shuffle` which have one argument `ary` and
+// returns a shuffled copy of `ary`
+// https://stackoverflow.com/questions/2450954/how-to-randomize-shuffle-a-javascript-array
+
+function shuffle(ary) {
+	let ary_shuffle = []
+	let ary_length = ary.length
+	for (let i = 0; i < ary_length; i++) {
+		let ran_num = Math.floor(Math.random() * (ary.length))
+		ary_shuffle.push(ary[ran_num])
+		ary.splice(ran_num, 1)
```

Comment:
we shouldn't use splice in `ary` since it will change `ary`'s content. The question required a copy of `ary` which mean `ary` should not changed after shuffle.